client/homescreen/activity-panel/test/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import ActivityPanel from '../';
+
+// Mock the panels.
+jest.mock( '../panels', () => {
```

Comment:
This mock reveals a code smell in the implementation of `<ActivityPanel>`. I can see why you're doing it, because if someone adds more rows then the test will fail, but it shows that you're using a private method inside `<ActivityPanel>` to render panels and that they should be a separate component.

Here's what I suggest:

1. Lets move away from the pattern of creating new objects like these panel objects which actually just represents what should be a component. Create a new `<Panel>` component.

A better panel component would be something like:

```
export const Panel = ({title, count, initialOpen, children}) => {

  // ... render the panel here
}
```

Some notes about why:

* every time we make a private method to render children in a component we remove a possible component we can reuse elsewhere. In the implementation of `<ActivityPanel>` as it is now, it cannot be reused and nor can any code used to generate the panels.

* Testing is easier, no mocking needed, test each very simple component separately.

* Note that increasing reuse will over time reduce bundle size because I imagine we're gonna reuse this UI pattern

2. With the new panel component in place you can now have `<ActivityPanel>` render it's children instead. Again making that component more flexible and removing the need for inlining component declarations in objects like we do in `getAllPanels`

3. Once `<ActivityPanel>` renders children we can write ActivityPanel like:

```
<ActivityPanel>
   {theData.map((whatever) => <Panel {...whatever} />)}
</ActivityPanel>
```

Doesn't look like much but in doing so we have decoupled the panel data from `ActivityPanel` (its just UI it doesn't need to know about the *data*.) And we made all the components fully reusable for elsewhere.

Reply:
This structure was changed in the [PR 5474](https://github.com/woocommerce/woocommerce-admin/pull/5474)